plugin/src/main/java/io/jenkins/plugins/casc/SchemaGeneration.java

```diff
@@ -0,0 +1,185 @@
+                                            .put("properties", attributeSchema));
+                            }
+                        }
+                    }
+                }
+            }
+            /**
+             * Used to generate the schema for the implementors of
+             * the HetroDescribable Configurator
+             * It mimics the HetroDescribable Configurator.jelly
+             */
+            else if (configuratorObject instanceof HeteroDescribableConfigurator) {
+                HeteroDescribableConfigurator heteroDescribableConfigurator = (HeteroDescribableConfigurator) configuratorObject;
+                schemaConfiguratorObjects.put(heteroDescribableConfigurator.getTarget().getSimpleName().toLowerCase(),
+                    generateHetroDescribableConfigObject(heteroDescribableConfigurator));
+                }
+            }
+        schemaObject.put("properties", schemaConfiguratorObjects);
+        return schemaObject;
+    }
+
+    private static JSONObject generateHetroDescribableConfigObject(HeteroDescribableConfigurator heteroDescribableConfiguratorObject) {
+        Map<String, Class> implementorsMap = heteroDescribableConfiguratorObject
+            .getImplementors();
+        JSONObject finalHetroConfiguratorObject = new JSONObject();
+        if (implementorsMap.size() != 0) {
+            Iterator<Map.Entry<String, Class>> itr = implementorsMap.entrySet().iterator();
+
+            JSONArray oneOfJsonArray = new JSONArray();
+            while (itr.hasNext()) {
+                Map.Entry<String, Class> entry = itr.next();
+                JSONObject implementorObject = new JSONObject();
+                implementorObject.put("properties",
+                    new JSONObject().put(entry.getKey(), new JSONObject()
+                        .put("$id", "#/definitions/" + entry.getValue().getName())));
+                oneOfJsonArray.put(implementorObject);
+            }
+
+            finalHetroConfiguratorObject.put("type", "object");
+            finalHetroConfiguratorObject.put("oneOf", oneOfJsonArray);
+        }
+            return finalHetroConfiguratorObject;
+    }
+
+    private static JSONObject generateNonEnumAttributeObject(Attribute attribute) {
+        JSONObject attributeType = new JSONObject();
+        switch (attribute.type.getName()) {
+            case "java.lang.String":
+                attributeType.put("type", "string");
+                break;
+
+            case "int":
+                attributeType.put("type", "integer");
+                break;
+
+            case "boolean":
+                attributeType.put("type", "boolean");
+                break;
+
+            case "java.lang.Boolean":
+                attributeType.put("type", "boolean");
+                break;
+
+            case "java.lang.Integer":
+                attributeType.put("type", "integer");
+                break;
+
+            case "java.lang.Long":
```

Comment:
`hudson.Secret` could be referenced as `string` in the schema

Reply:
Yeah, I will include it in the switch case :)